File: src/model/Profession.java
package model;

import model.item.ItemType;

/**
 * Professions with their gold/score multiplier and starting skill
 * 
 */
public enum Profession {
	BANKER (1, Skill.COMMERCE, "Banker", null),
	DOCTOR (1.2f, Skill.MEDICAL, "Doctor", null),
	MERCHANT (1.4f, Skill.COMMERCE, "Merchant", null),
	PHARMACIST (1.6f, Skill.MEDICAL, "Pharmacist", null),
	WAINWRIGHT (1.8f, Skill.CARPENTRY, "Wainwright", ItemType.WHEEL),
	GUNSMITH (2, Skill.SHARPSHOOTING, "Gunsmith", ItemType.GUN),
	BLACKSMITH (2.2f, Skill.BLACKSMITHING, "Blacksmith", null),
	MASON (2.4f, Skill.CARPENTRY, "Mason", null),
	WHEELWRIGHT (2.6f, Skill.CARPENTRY, "Wheelwright", ItemType.WHEEL),
	CARPENTER (2.8f, Skill.CARPENTRY, "Carpenter", ItemType.WHEEL),
	SADDLEMAKER (3, Skill.FARMING, "Saddlemaker", null),
	BRICKMAKER (3.2f, Skill.CARPENTRY, "Brickmaker", null),
	PROSPECTOR (3.4f, Skill.RIVERWORK, "Prospector", null),
	TRAPPER (3.6f, Skill.TRACKING, "Trapper", null),
	SURVEYOR (3.8f, Skill.RIVERWORK, "Surveyor", null),
	SHOEMAKER (4, Skill.SEWING, "Shoemaker", null),
	JOURNALIST (4.1f, Skill.NONE, "Journalist", null),
	PRINTER (4.2f, Skill.COMMERCE, "Printer", null),
	BUTCHER (4.3f, Skill.COOKING, "Butcher", ItemType.MEAT),
	BAKER (4.4f, Skill.COOKING, "Baker", ItemType.BREAD),
	TAILOR (4.5f, Skill.SEWING, "Tailor", null),
	FARMER (4.5f, Skill.FARMING, "Farmer", ItemType.APPLE),
	PASTOR (4.6f, Skill.NONE, "Pastor", null),
	ARTIST (4.8f, Skill.MUSICAL, "Artist", null),
	TEACHER (5, Skill.NONE, "Teacher", null);
	
	private final float moneyDivider;

	private final Skill startingSkill;
	
	private final String name;
	
	private final ItemType startingItem;
	
	private static final float BASE_MONEY = 1600f;
	
	/**
	 * Assigns money multiplier and starting skill to each profession
	 * @param moneyDivider Gold divided by and score multiplied by this 
	 * @param startingSkill Skill the profession starts with
	 * @param name The profession's name
	 */
	private Profession(float moneyDivider, Skill startingSkill, String name, ItemType itemType) {
		this.moneyDivider = moneyDivider;
		this.startingSkill = startingSkill;
		this.name = name;
		this.startingItem = itemType;
	}
	
	/**
	 * Returns the multiplier
	 * @return Multiplier
	 */
	public int getMoney() {
		return (int) (BASE_MONEY / moneyDivider);
	}
	
	/**
	 * Returns starting skill
	 * @return Starting skill
	 */
	public Skill getStartingSkill() {
		return startingSkill;
	}
	
	public ItemType getStartingItem() {
		return startingItem;
	}
	
	/**
	 * Returns the profession's name
	 * @return Profession's name
	 */
	public String getName(){
		return name;
	}
		
	public String toString() {
		return name;
	}
}

File: web/src/model/Profession.ts
import { Skill } from "./Skill";

export enum Profession {
  BANKER = "BANKER",
  DOCTOR = "DOCTOR",
  MERCHANT = "MERCHANT",
  PHARMACIST = "PHARMACIST",
  WAINWRIGHT = "WAINWRIGHT",
  GUNSMITH = "GUNSMITH",
  BLACKSMITH = "BLACKSMITH",
  MASON = "MASON",
  WHEELWRIGHT = "WHEELWRIGHT",
  CARPENTER = "CARPENTER",
  SADDLEMAKER = "SADDLEMAKER",
  BRICKMAKER = "BRICKMAKER",
  PROSPECTOR = "PROSPECTOR",
  TRAPPER = "TRAPPER",
  SURVEYOR = "SURVEYOR",
  SHOEMAKER = "SHOEMAKER",
  JOURNALIST = "JOURNALIST",
  PRINTER = "PRINTER",
  BUTCHER = "BUTCHER",
  BAKER = "BAKER",
  TAILOR = "TAILOR",
  FARMER = "FARMER",
  PASTOR = "PASTOR",
  ARTIST = "ARTIST",
  TEACHER = "TEACHER",
}

const BASE_MONEY = 1600;

interface ProfessionData {
  moneyDivider: number;
  startingSkill: Skill;
  name: string;
  startingItem: string | null;
}

const PROFESSION_DATA: Record<Profession, ProfessionData> = {
  [Profession.BANKER]: { moneyDivider: 1, startingSkill: Skill.COMMERCE, name: "Banker", startingItem: null },
  [Profession.DOCTOR]: { moneyDivider: 1.2, startingSkill: Skill.MEDICAL, name: "Doctor", startingItem: null },
  [Profession.MERCHANT]: { moneyDivider: 1.4, startingSkill: Skill.COMMERCE, name: "Merchant", startingItem: null },
  [Profession.PHARMACIST]: { moneyDivider: 1.6, startingSkill: Skill.MEDICAL, name: "Pharmacist", startingItem: null },
  [Profession.WAINWRIGHT]: { moneyDivider: 1.8, startingSkill: Skill.CARPENTRY, name: "Wainwright", startingItem: "WHEEL" },
  [Profession.GUNSMITH]: { moneyDivider: 2, startingSkill: Skill.SHARPSHOOTING, name: "Gunsmith", startingItem: "GUN" },
  [Profession.BLACKSMITH]: { moneyDivider: 2.2, startingSkill: Skill.BLACKSMITHING, name: "Blacksmith", startingItem: null },
  [Profession.MASON]: { moneyDivider: 2.4, startingSkill: Skill.CARPENTRY, name: "Mason", startingItem: null },
  [Profession.WHEELWRIGHT]: { moneyDivider: 2.6, startingSkill: Skill.CARPENTRY, name: "Wheelwright", startingItem: "WHEEL" },
  [Profession.CARPENTER]: { moneyDivider: 2.8, startingSkill: Skill.CARPENTRY, name: "Carpenter", startingItem: "WHEEL" },
  [Profession.SADDLEMAKER]: { moneyDivider: 3, startingSkill: Skill.FARMING, name: "Saddlemaker", startingItem: null },
  [Profession.BRICKMAKER]: { moneyDivider: 3.2, startingSkill: Skill.CARPENTRY, name: "Brickmaker", startingItem: null },
  [Profession.PROSPECTOR]: { moneyDivider: 3.4, startingSkill: Skill.RIVERWORK, name: "Prospector", startingItem: null },
  [Profession.TRAPPER]: { moneyDivider: 3.6, startingSkill: Skill.TRACKING, name: "Trapper", startingItem: null },
  [Profession.SURVEYOR]: { moneyDivider: 3.8, startingSkill: Skill.RIVERWORK, name: "Surveyor", startingItem: null },
  [Profession.SHOEMAKER]: { moneyDivider: 4, startingSkill: Skill.SEWING, name: "Shoemaker", startingItem: null },
  [Profession.JOURNALIST]: { moneyDivider: 4.1, startingSkill: Skill.NONE, name: "Journalist", startingItem: null },
  [Profession.PRINTER]: { moneyDivider: 4.2, startingSkill: Skill.COMMERCE, name: "Printer", startingItem: null },
  [Profession.BUTCHER]: { moneyDivider: 4.3, startingSkill: Skill.COOKING, name: "Butcher", startingItem: "MEAT" },
  [Profession.BAKER]: { moneyDivider: 4.4, startingSkill: Skill.COOKING, name: "Baker", startingItem: "BREAD" },
  [Profession.TAILOR]: { moneyDivider: 4.5, startingSkill: Skill.SEWING, name: "Tailor", startingItem: null },
  [Profession.FARMER]: { moneyDivider: 4.5, startingSkill: Skill.FARMING, name: "Farmer", startingItem: "APPLE" },
  [Profession.PASTOR]: { moneyDivider: 4.6, startingSkill: Skill.NONE, name: "Pastor", startingItem: null },
  [Profession.ARTIST]: { moneyDivider: 4.8, startingSkill: Skill.MUSICAL, name: "Artist", startingItem: null },
  [Profession.TEACHER]: { moneyDivider: 5, startingSkill: Skill.NONE, name: "Teacher", startingItem: null },
};

export function getProfessionMoney(profession: Profession): number {
  return Math.floor(BASE_MONEY / PROFESSION_DATA[profession].moneyDivider);
}

export function getProfessionStartingSkill(profession: Profession): Skill {
  return PROFESSION_DATA[profession].startingSkill;
}

export function getProfessionName(profession: Profession): string {
  return PROFESSION_DATA[profession].name;
}

export function getProfessionStartingItem(profession: Profession): string | null {
  return PROFESSION_DATA[profession].startingItem;
}
